fixham/internal/version/version.go

```diff
@@ -0,0 +1,79 @@
+package version
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+
+	"github.com/projectcalico/calico/fixham/internal/calico"
+)
+
+type Version string
+
+func (v *Version) Set(value string) error {
```

Comment:
This function is never called - is it needed?

It's a pretty odd pattern to have unless there's really a use-case for setting a struct. Typically something like being hyper memory efficient.

I think this probably wants a package-level New() function instead.

```go
func New(val string) Version {
  // Validate, then cast as a Version().
  return Version(val)
}
```